hbase-common/src/main/java/org/apache/hadoop/hbase/util/CommonFSUtils.java

```diff
@@ -350,11 +350,18 @@ public static void setWALRootDir(final Configuration c, final Path root) {
   public static FileSystem getWALFileSystem(final Configuration c) throws IOException {
     Path p = getWALRootDir(c);
     FileSystem fs = p.getFileSystem(c);
-    // hadoop-core does fs caching, so need to propogate this if set
+    // hadoop-core does fs caching, so need to propagate this if set
     String enforceStreamCapability = c.get(UNSAFE_STREAM_CAPABILITY_ENFORCE);
     if (enforceStreamCapability != null) {
       fs.getConf().set(UNSAFE_STREAM_CAPABILITY_ENFORCE, enforceStreamCapability);
     }
+    if (!c.getBoolean(HConstants.CLUSTER_DISTRIBUTED, false)) {
```

Comment:
nit: make it an else if? meaning if the user didn't override UNSAFE_STREAM_CAPABILITY_ENFORCE and its a local fs mode..

Reply:
Can you not ask fs if it is an instance of local fs? If it is, then set the boolean?

Reply:
Hmm.... Looking here, the distributed flag is good to check [1] but would we run non-distributed on an fs that supports sync? s3?

1. http://hbase.apache.org/book.html#standalone

Reply:
> make it an else if? meaning if the user didn't override UNSAFE_STREAM_CAPABILITY_ENFORCE and its a local fs mode..

I explicitly disable it here because we know it doesn't work with the local filesystem. There are plenty of places where conf lookups of `UNSAFE_STREAM_CAPABILITY_ENFORCE` have a default of `true`, so leaving it unset sometimes doesn't help.

Reply:
> Can you not ask fs if it is an instance of local fs? If it is, then set the boolean?
> would we run non-distributed on an fs that supports sync? s3?

My preference would be `if !distributed && !fs.supportsCapabilities()`, however the only way I see to ask a FS if it supports the capabilities is to first instantiate a stream, which I didn't want to do here. Other suggestions?

Reply:
I think @saintstack is right. We can run non-distributed on a distributed fs with stream capabilities. So the check is probably not tight...

> Other suggestions?

One naive way is to just check for instance of LocalFileSystem and override? (Think this is what Stack was suggesting). This covers the most common case of testing something by just unpacking a bin release. Any other fancy filesystems, users may have to figure out themselves?

Reply:
I was hoping to avoid pinning the logic to specific `FileSystem` implementations, rather on capabilities. Fun fact: the only commonality between `LocalFileSystem` and `RawLocalFileSystem` is `FileSystem` :/

Reply:
Ya, that'd be ideal.. probably a new Hadoop jira for the FileSystem interface to expose the capabilities in some form...

Reply:
From debugging, it's `LocalFileSystem` that gets used for wals, so that's what I've used. This seems very fragile though :/